src/main/java/build/buildfarm/server/BuildFarmServer.java

```diff
@@ -107,40 +114,67 @@ public BuildFarmServer(
-            .addService(
-                new ExecutionService(
-                    instance,
-                    config.getExecuteKeepaliveAfterSeconds(),
-                    TimeUnit.SECONDS,
-                    keepaliveScheduler,
-                    getMetricsPublisher(config.getMetricsConfig())))
-            .addService(new OperationQueueService(instance))
-            .addService(new OperationsService(instance))
-            .addService(new AdminService(config.getAdminConfig(), instance))
-            .addService(new FetchService(instance))
-            .addService(ProtoReflectionService.newInstance())
-            .addService(new PublishBuildEventService(config.getBuildEventConfig()))
-            .intercept(TransmitStatusRuntimeExceptionInterceptor.instance())
-            .intercept(headersInterceptor)
-            .build();
 
-    logger.log(Level.INFO, String.format("%s initialized", session));
+    serverBuilder.addService(healthStatusManager.getHealthService());
+    serverBuilder.addService(new ActionCacheService(instance, config.getAcPolicy()));
+    serverBuilder.addService(new CapabilitiesService(instance));
+    serverBuilder.addService(
+        new ContentAddressableStorageService(
+            instance, /* deadlineAfter=*/ config.getCasWriteTimeout().getSeconds(), TimeUnit.SECONDS
+            /* requestLogLevel=*/ ));
+    serverBuilder.addService(
+        new ByteStreamService(
+            instance,
+            /* writeDeadlineAfter=*/ config.getBytestreamTimeout().getSeconds(),
+            TimeUnit.SECONDS));
+    serverBuilder.addService(
+        new ExecutionService(
+            instance,
+            config.getExecuteKeepaliveAfterSeconds(),
+            TimeUnit.SECONDS,
+            keepaliveScheduler,
+            getMetricsPublisher(config.getMetricsConfig())));
+    serverBuilder.addService(new OperationQueueService(instance));
+    serverBuilder.addService(new OperationsService(instance));
+    serverBuilder.addService(new AdminService(config.getAdminConfig(), instance));
+    serverBuilder.addService(new FetchService(instance));
+    serverBuilder.addService(ProtoReflectionService.newInstance());
+    serverBuilder.addService(new PublishBuildEventService(config.getBuildEventConfig()));
+
+    serverBuilder.intercept(TransmitStatusRuntimeExceptionInterceptor.instance());
+    serverBuilder.intercept(headersInterceptor);
+    handleGrpcMetricIntercepts(serverBuilder, config);
+
+    return serverBuilder.build();
+  }
+
+  private static void handleGrpcMetricIntercepts(
+      ServerBuilder<?> serverBuilder, BuildFarmServerConfig config) {
+    // Decide how to capture GRPC Prometheus metrics.
+    // By default, we don't capture any.
+    if (config.getGrpcMetrics().getEnabled()) {
+      // Assume cheap metrics.
+      // Cheap metrics include send/receive totals tagged with return codes.  No latencies.
+      Configuration grpcConfig = Configuration.cheapMetricsOnly();
```

Comment:
nit: "cheap" should probably be replaced by something like 'core'. otherwise, you may want to flip the config, and add an option like `include_latencies`. If we add more things, it'll be a clear pattern on how to add more flags in the future.

Reply:
Also, I dont see a cheap_metrics_only option. Looks like this function is missing unit tests.

Reply:
its a good point.  the "cheap" terminology came from the library itself.  
For our purposes, we could say that our core metrics are what this library is considering as default (which is what you get through the cheapMetricsOnly() call).  

If you enable the metrics with `grpc_metrics.enabled: true`, you'll get those core metrics.  
If you want more, you will further enable the hisograms with `grpc_metrics.provide_latency_histograms: true`.  

Trying to match how their [library config](https://github.com/grpc-ecosystem/java-grpc-prometheus/blob/8b4d92e76f0f59bf4393b3f8d6cdfdd1386154e3/src/main/java/me/dinowernli/grpc/prometheus/Configuration.java#L25) code works:  

```
  /** Returns a {@link Configuration} for recording all cheap metrics about the rpcs. */
  public static Configuration cheapMetricsOnly() {
    return new Configuration(
        false /* isIncludeLatencyHistograms */,
        CollectorRegistry.defaultRegistry,
        DEFAULT_LATENCY_BUCKETS,
        new ArrayList<>(),
        false /* isAddCodeLabelToHistograms */);
  }

  /**
   * Returns a {@link Configuration} for recording all metrics about the rpcs. This includes metrics
   * which might produce a lot of data, such as latency histograms.
   */
  public static Configuration allMetrics() {
    return new Configuration(
        true /* isIncludeLatencyHistograms */,
        CollectorRegistry.defaultRegistry,
        DEFAULT_LATENCY_BUCKETS,
        new ArrayList<>(),
        false);
  }
```

You can see their idea of "cheap" metrics... or core metrics, is having the other metric options turned off.

Reply:
changing our use of "cheap" to say "core".

Reply:
Oh, my bad, I should have followed the object type to the top. Yeah, I see that cheap is something we may be stuck with, though I don't think this is a descriptive term.  

Also, how about that unit test for this static function? ;)

Reply:
added unit tests